server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapper.java

```diff
@@ -78,7 +78,14 @@ public RuleMapper(final Languages languages, final MacroInterpreter macroInterpr
   public Rules.Rule toWsRule(RuleDto ruleDto, SearchResult result, Set<String> fieldsToReturn) {
     Rules.Rule.Builder ruleResponse = Rules.Rule.newBuilder();
 
+    // Mandatory fields
     ruleResponse.setKey(ruleDto.getKey().toString());
+    Common.RuleType type = Common.RuleType.valueOf(ruleDto.getType());
+    if (type != null) {
```

Comment:
@teryk told me that only fields that are complex to load should be put in optional fields.
And so every "simple" fields should be in mandatory fields.
It's not the case ?